shapely/geometry/geo.py

```diff
@@ -11,13 +11,44 @@
 from .collection import GeometryCollection
 
 
+def _is_coordinates_empty(coordinates):
+    """Helper to identify if list or all nested lists of coordinates are empty"""
+    if isinstance(coordinates, (list, tuple, np.ndarray)):
+        if len(coordinates) == 0:
```

Comment:
To avoid making numpy a dependency here, what would you think about something like 

```suggestion
        if coordinates is None:
            return True
        else:
            try:
                coords_len = len(coords)
                ...
            except TypeError:
                return False
```

Reply:
Changes needed to following lines of course.